File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam20And21April2019/04.EasterEggsBattle.js
function easterEggsBattle(input) {
    let index = 0;
    let firstPlayerEggs = Number(input[index]);
    index++;
    let secondPlayerEggs = Number(input[index]);
    index++;
    let command = input[index];
    index++;

    while(command !== 'End') {
        let playerTurn = command;

        switch(playerTurn) {
            case 'one': secondPlayerEggs--; break;
            case 'two': firstPlayerEggs--; break;
        }

        if(firstPlayerEggs <= 0) {
            console.log(`Player one is out of eggs. Player two has ${secondPlayerEggs} eggs left.`);
            break;
        }

        if(secondPlayerEggs <= 0) {
            console.log(`Player two is out of eggs. Player one has ${firstPlayerEggs} eggs left.`);
            break;
        }

        command = input[index];
        index++;
    }

    if(firstPlayerEggs > 0 && secondPlayerEggs > 0) {
        console.log(`Player one has ${firstPlayerEggs} eggs left.`);
        console.log(`Player two has ${secondPlayerEggs} eggs left.`);
    }
}

easterEggsBattle(['5', '4', 'one', 'two', 'one', 'two', 'two', 'End'])
easterEggsBattle(['2', '6', 'one', 'two', 'two'])
easterEggsBattle(['6', '3', 'one', 'two', 'two', 'one', 'one', 'End'])

File: Programming_Basics_with_Python/programming_basics_online_exam_20_and_21_april_2019/04._easter_eggs_battle.py
first_player_eggs_count = int(input())
second_player_eggs_count = int(input())
command = input()

message = ''
no_winner = True
while not command == 'End':
    winner = command
    if winner == 'one':
        second_player_eggs_count -= 1
        if second_player_eggs_count == 0:
            no_winner = False
            message = f'Player two is out of eggs. Player one has {first_player_eggs_count} eggs left.'
            break
    elif winner == 'two':
        first_player_eggs_count -= 1
        if first_player_eggs_count == 0:
            no_winner = False
            message = f'Player one is out of eggs. Player two has {second_player_eggs_count} eggs left.'
            break

    command = input()

if no_winner:
    print(f'Player one has {first_player_eggs_count} eggs left.')
    print(f'Player two has {second_player_eggs_count} eggs left.')
else:
    print(message)